File: src/main/java/com/antlr/plugin/configdialogs/CaseChangingStrategyConverter.java
package com.antlr.plugin.configdialogs;

import com.intellij.util.xmlb.Converter;
import com.antlr.plugin.parsing.CaseChangingStrategy;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * A converter used to serialize/deserialize a {@code CaseChangingStrategy} to/from
 * a String contained in {@code .idea/misc.xml} using the enum's {@code name()}.
 */
public class CaseChangingStrategyConverter extends Converter<CaseChangingStrategy> {
    @Nullable
    @Override
    public CaseChangingStrategy fromString(@NotNull String value) {
        try {
            return CaseChangingStrategy.valueOf(value);
        } catch (IllegalArgumentException e) {
            return null;
        }
    }

    @Nullable
    @Override
    public String toString(@NotNull CaseChangingStrategy value) {
        return value.name();
    }
}

File: src/main/java/com/antlr/setting/configdialogs/CaseChangingStrategyConverter.kt
package com.antlr.setting.configdialogs

import com.antlr.language.psrsing.CaseChangingStrategy
import com.intellij.util.xmlb.Converter

class CaseChangingStrategyConverter : Converter<CaseChangingStrategy>() {
    override fun fromString(value: String): CaseChangingStrategy? {
        try {
            return CaseChangingStrategy.valueOf(value)
        } catch (e: IllegalArgumentException) {
            return null
        }
    }

    override fun toString(caseChangingStrategy: CaseChangingStrategy): String? {
        return caseChangingStrategy.name
    }
}
